das/factory.go

```diff
@@ -55,3 +59,54 @@ func CreatePersistentStorageService(
 	}
 	return nil, &lifecycleManager, nil
 }
+
+func CreateBatchPosterDAS(
+	ctx context.Context,
+	config *DataAvailabilityConfig,
+	daSigner DasSigner,
+	l1Reader arbutil.L1Interface,
+	sequencerInboxAddr common.Address,
+) (DataAvailabilityServiceWriter, DataAvailabilityServiceReader, *LifecycleManager, error) {
+	if !config.Enable {
+		return nil, nil, nil, nil
+	}
+
+	if !config.AggregatorConfig.Enable || !config.RestfulClientAggregatorConfig.Enable {
+		return nil, nil, nil, errors.New("--node.data-availabilty.rpc-aggregator.enable and rest-aggregator.enable must be set when running a Batch Poster in AnyTrust mode.")
+	}
+
+	if config.LocalDBStorageConfig.Enable || config.LocalFileStorageConfig.Enable || config.S3StorageServiceConfig.Enable {
+		return nil, nil, nil, errors.New("--node.data-availability.local-db-storage.enable, local-file-storage.enable, s3-storage.enable may not be set when running a Batch Poster in AnyTrust mode.")
+	}
+
+	if config.KeyConfig.KeyDir != "" || config.KeyConfig.PrivKey != "" {
+		return nil, nil, nil, errors.New("--node.data-availability.key.key-dir, priv-key not be set when running a Batch Poster in AnyTrust mode.")
```

Comment:
I think these checks are probably flipped

Reply:
If the BLS signing key is specified in either way, then error because this shouldn't be set on the batch poster.

Reply:
Had gotten confused here. Makes sense